configen/FileGenerator.swift

```diff
@@ -94,35 +94,72 @@ struct FileGenerator {
     }
     
     var line = ""
+    var finalType = type
+    var isObjcOptional = false
+    
+    if optionsParser.isObjC && type.contains("?") {
+      finalType = type.replacingOccurrences(of: "?", with: "")
+      isObjcOptional = value as? String == "nil" ? true : false
+    }
     
-    switch (type) {
+    switch (finalType) {
     case ("Double"):
-      line = template.doubleImplementation
+      if isObjcOptional && template is ObjectiveCTemplate{
+        let objcTemplate = template as! ObjectiveCTemplate
+        line = objcTemplate .optionalImplementation
+        line.replace(token: objcTemplate.declarationToken, withString: objcTemplate.doubleDeclaration)
+      }else {
+        line = template.doubleImplementation
+      }
       
     case ("Int"):
-      line = template.integerImplementation
+      if isObjcOptional && template is ObjectiveCTemplate{
```

Comment:
could this be if let objcTemplate = template as? ObjectiveCTemplate, isObjcOptional == true? to avoid the force cast after?